### ISSUE TITLE
Connection Errors: Error response from daemon

### ISSUE THREAD

**Issue by eczajk1** [docker/for-win#578]
### Expected behavior

To be able to run simple docker commands like ```login``` and ```build```.

I recently updated to my current version of Docker.

### Actual behavior

Not able to get a connection.  Error messages:

```sh
$ docker login
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username: myusername
Password:
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

```sh
$ docker build -t build-this .
Sending build context to Docker daemon 37.13 MB
Step 1/8 : FROM node:argon
Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection
```

### Information

  - Diagnostic ID from "Diagnose & Feedback" in the menu.
    - A6F111CD-5243-4350-A91D-32B774D556AB/2017-03-13_10-31-43
  - ~a reproducible case if this is a bug, Dockerfiles FTW~
  - ~page URL if this is a docs issue or the name of a man page~
  - host distribution and version (Windows version, build number, etc)
    - Version 17.03.0-ce-win1 (10296)
    - Channel: stable
    - e5a07a1

### Steps to reproduce the behavior

Not trying to do anything crazy here:
  1. I was just prompted to update to the new version of Docker.
  2. Then, I rebooted my machine.
  3. The rest is shown in the "Actual Behavior" section (above)

**Comment by th-ko** [comment]
Same problem with Docker version 17.03.0-ce, build 60ccb22

**Comment by eczajk1** [comment]
fyi, i was having this problem yesterday - today it doesn't seem to be a problem. 😖

**Comment by kotovaleksandr** [comment]
With Version 17.03.0-ce-win1 (10300) have this problem

**Comment by th-ko** [comment]
Seems the issue was related to an invalid secondary DNS-server, solved now

**Comment by eczajk1** [comment]
Does that mean it should work now, or will that take some time to propogate?  I'm still experiencing an issue:

```sh
$ docker login
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username: myusername
Password:
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection
```

**Comment by eczajk1** [comment]
Here's another message that I've been seeing repeatedly today:

```sh
Sending build context to Docker daemon 114.3 MB
Step 1/18 : FROM ubuntu:trusty
Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection
```

**Comment by th-ko** [comment]
@eczajk1  Well, in my case the problem was the configuration of my network-interface. It was set up to an non-existing secondary DNS-server. Our DNS-server are company internal.

**Comment by kotovaleksandr** [comment]
@th-ko @eczajk1 Yes, i solved this issue after set actual dns servers in DockerNAT network connection.

**Comment by fharris** [comment]
@th-ko, @eczajk1, @kotovaleksandr  2 sorry guys... I'm running in the same issue with Ubuntu this morning. How did you manage to sort this out ? Where and how can I make those DNS changes ?? Thank you very much.

**Comment by TSPcoder** [comment]
Same problem here : 
When I try `$ docker run hello-world`, I get the following error :


`$ docker run hello-world
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).`

**Information :**

- Docker toolbox used
- Docker version 17.05.0-ce, build 89658be
- Windows 7 Professional

I am running docker behind a **company proxy** (I have set the environment variables HTTP_PROXY and HTTPS_PROXY in the following file :  /var/lib/boot2docker/profile)

I also set those environment variables in the docker toolbox shell as illustrated by this command : 
`$ printenv | grep PROXY
NO_PROXY=192.168.99.101
HTTPS_PROXY=myCompanyProxy:myPortNumber
HTTP_PROXY=myCompanyProxy:myPortNumber`

I have this problem even though I should have had the same problem when docker tried to download the boot2docker iso file when attempting to create the default machine for the first time (I know this because I had this problem).

However, when creating the docker default machine with the command below, it succeeded in dowloading the boot2docker iso file to create the virtual machine :

`$ docker-machine create -d virtualbox \
--engine-env HTTP_PROXY=http://myCompanyProxy:myPortNumber:8080 \
--engine-env HTTPS_PROXY=http://myCompanyProxy:myPortNumber:8080 \
--virtualbox-no-vtx-check default`

I think the problem come from the DNS as you just said, however I don't know where I can make the change. Can someone help me ?

Thank you

**Comment by friism** [comment]
@TSPcoder please open an issue on Docker Toolbox if there's not already one there: https://github.com/docker/toolbox/

**Comment by DaniGTA** [comment]
I have the same error on windows 10. With dns on interface 1.1.1.1 +  dns setting in docker deamon:

`
{
  "builder": {
    "gc": {
      "defaultKeepStorage": "20GB",
      "enabled": true
    }
  },
  "dns": [
    "1.1.1.1",
    "8.8.8.8"
  ],
  "experimental": false
}
`

`docker info`
Client:
 Version:    25.0.3
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1-desktop.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.6-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.24
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.22
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.0.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.5.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 15
  Running: 15
  Paused: 0
  Stopped: 0
 Images: 18
 Server Version: 25.0.3
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.133.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 5.653GiB
 Name: docker-desktop
 ID: 0f4b3361-ebfe-489b-90ed-b4244a090c11
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
